app/javascript/components/set-ownership-form/index.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from 'patternfly-react';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import { http, API } from '../../http_api';
+import createSchema from './ownership-form.schema';
+
+class SetOwnershipForm extends Component {
+  state = {
+    initialValues: {},
+    userOptions: [],
+    groupOptions: [],
+  };
+
+  componentDidMount() {
+    const { ownershipIds } = this.props;
+    this.loadInitialData(ownershipIds);
+  }
+
+  /**
+   * Adds neutral options: "Don't Change" and "No Owner"/"No User Group" to the select box options.
+   *
+   * @param ownershipIds Array of element ids being edited in the form. If more than one add "Don't Change" option.
+   * @param type The type of the options to which elements are currently being added. ('user' or 'group')
+   * @param data The data already in the select box. To this the function prepends the new fields.
+   * @param tenantGroupId If the type is 'group' and the ownershipIds.length is one then the value of the 'No User Group' field will be this value.
+   * @returns {*[]}
+   */
+  addOptions = (ownershipIds, type, data, tenantGroupId = '') => {
+    const ret = [];
+    if (ownershipIds.length > 1) {
+      ret.push([__('Don\'t change'), 'dont-change']);
+    }
+    ret.push([type === 'user' ? __('No Owner') : __('No User Group'), ownershipIds.length === 1 ? tenantGroupId : '']);
+
+    return [...ret, ...data];
+  };
+
+  loadInitialData = (ownershipIds) => {
+    Promise.all([
+      API.get('/api/users?expand=resources&attributes=id,name&sort_by=name&sort_order=ascending'),
+      API.get('/api/groups?expand=resources&attributes=id,description&sort_by=description&sort_order=ascending'),
+      http.post(`/${ManageIQ.controller}/ownership_form_fields`, { object_ids: ownershipIds }),
+    ])
+      .then(([userOptions, groupOptions, initialValues]) => {
+        // Checking if the group is tenant group or not
+        const tenantGroupId = new Promise((resolve, reject) => {
+          API.get(`/api/tenant_groups/${initialValues.group}`, { skipErrors: true })
+            .then((response) => { resolve(response.id); })
+            .catch(() => resolve(''));
+        });
```

Comment:
Nope, this is still not okay. You need something like this:
```
Promise.all().then(/* test tenant group */, /* same as the catch */).then(/* setState */)
```

Reply:
The first `then` has the `catch` inside as the second argument, look [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#Syntax) for syntax.